picard/coverart/providers/__init__.py

```diff
@@ -120,15 +136,11 @@ def __init__(self, coverart):
         self.metadata = coverart.metadata
         self.album = coverart.album
 
-    def enabled(self):
-        """By default, return True if user enabled the provider
-        through options. It is used when iterating through providers
-        to decide to skip or process one.
-        It can be subclassed to add conditions."""
-        enabled = is_provider_enabled(self.NAME)
-        if not enabled:
-            log.debug("%s disabled by user" % self.NAME)
-        return enabled
+    @staticmethod
+    def enabled():
+        # kept for compatibility
+        # deprecated, and always returns True
+        return True
```

Comment:
The comment seems to be wrong, as this is still in use in derived classes. There are a few cases where this still can be useful that a provider can decide to be skipped, e.g. the Fanart.tv plugin makes use of this to disable itself if no API key is available. Also the CAA provider has some additional checks.

On the other hand all current implementations check something like `return not self.coverart.front_image_found`, this seems unnecessary. That's something the `CoverArt` class could already take care of.

Reply:
One use case for not checking `self.coverart.front_image_found` would be if the plugin is meant to download additional images other than front, and you might want to do this always. E.g. Fanart.tv and TheAudioDb plugins currently can also download media (CD) images. They don't make use of this yet, but one scenario could be that they can always download the CD image even if a front image has already been found.

So I would actually suggest to change the default implementation here to 

```
def enabled(self):
        return not self.coverart.front_image_found
```

And only override this in sub classes if necessary (and usually call `super().enabled()`). That way implementations can make easy use of the default, hut if necessary still can override the check for the front image.

Reply:
Done in 56466b9433f15e8

Reply:
Yes, this looks  even cleaner now.